test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs

```diff
@@ -30,37 +30,75 @@ public class GivenThatWeWantToBuildANetCoreApp : SdkTest
         [InlineData("netcoreapp1.1", null, "1.1.2", "1.1.2")]
         [InlineData("netcoreapp1.1", "1.1.0", "1.1.0", "1.1.0")]
         [InlineData("netcoreapp1.1.1", null, "1.1.1", "1.1.1")]
-        public void It_targets_the_right_shared_framework(string targetFramework, string runtimeFrameworkVersion,
+        private void It_targets_the_right_shared_framework(string targetFramework, string runtimeFrameworkVersion,
             string expectedPackageVersion, string expectedRuntimeVersion)
         {
+            string testIdentifier = "SharedFrameworkTargeting_" + string.Join("_", targetFramework, runtimeFrameworkVersion ?? "null");
+
+            It_targets_the_right_framework(testIdentifier, targetFramework, runtimeFrameworkVersion,
+                selfContained: false, isExe: true,
+                expectedPackageVersion: expectedPackageVersion, expectedRuntimeVersion: expectedRuntimeVersion);
+        }
+
+        //  Test behavior when implicit version differs for shared framework and self-contained apps
+        [Theory]
+        [InlineData(false, true, "1.1.1")]
+        [InlineData(true, true, "1.1.2")]
+        [InlineData(false, false, "1.1.1")]
+        public void It_targets_the_right_framework_depending_on_output_type(bool selfContained, bool isExe, string expectedFrameworkVersion)
+        {
+            string testIdentifier = "Framework_targeting_" + (isExe ? "App_" : "Lib_") + (selfContained ? "SelfContained" : "SharedFramework");
+
+            It_targets_the_right_framework(testIdentifier, "netcoreapp1.1", null, selfContained, isExe, expectedFrameworkVersion, expectedFrameworkVersion,
+                "/p:ImplicitRuntimeFrameworkVersionForSharedNetCoreApp1_1=1.1.1");
+        }
+
+        private void It_targets_the_right_framework(
+            string testIdentifier,
+            string targetFramework, string runtimeFrameworkVersion,
```

Comment:
NIT: I think putting some args on new lines, and some inline makes this hard to read.